modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js

```diff
@@ -227,20 +244,38 @@ const RichText = ({
 	}
 
 	const resetTranslation = useCallback(() => {
-		editorRef.current.editor.setData(currentValue[defaultLocale.localeId]);
-	}, [editorRef, currentValue, defaultLocale]);
+		const data = currentValue[defaultLocale.localeId];
+
+		if (Liferay.FeatureFlags['LPD-11235']) {
+			setCKEditor5Config({
+				...ckEditor5Config,
+				initialData: data ?? '',
+			});
+		}
+		else {
+			editorRef.current.editor.setData(data);
+		}
+	}, [ckEditor5Config, currentValue, defaultLocale, editorRef]);
 
 	useEffect(() => {
 		const handleRestoreState = () => {
-			editorRef.current.editor.setData(value);
+			if (Liferay.FeatureFlags['LPD-11235']) {
+				setCKEditor5Config({
+					...ckEditor5Config,
+					initialData: value,
+				});
```

Comment:
this looks a much nicer way to set data into the editor, as compared to `editorRef.current.editor.setData(value)`

Reply:
I explored the idea of doing this in a similar way. We can expose `editor` object in `onReady`, and `setData` on it. That's the standard API, that can be better good solution in some other usage. The problem here is that `ref` is not stable, because we are destroying and recreating component, so I found cases where the feature would not work. By doing this, we recreate component even in this case, so there is no issue.